docs/proposals/05-etcd-operator-tasks.md

```diff
@@ -0,0 +1,482 @@
+  LastOps []LastOperation
+}
+
+```
+
+**Note**: If both type of maintenance is requested then controller should trigger etcd-compaction first then defragmentation.
+
+### Dealing with multiple operator tasks simultaneously
+
+It is quite apparent that multiple "out-of-band" operator tasks can be triggered simultaneously. The challenge then becomes determining the precedence of each operation.
+For instance, let's say an operator triggers an on-demand full snapshot task that doesn't complete. Then, the human operator realizes the cluster has lost its quorum, so they trigger a `recover-quorum-loss` task.
+In this scenario, our controller will detect two pending tasks at the same time. The question that arises is: which task should be executed first?
+
+To deal with such scenario, authors are proposing that each operator out-of-bands task will divided mainly into 2 priority class:
+
+1. Cluster-ops
+2. Backup-ops
+
+Assignment of different tasks to priority class is mentioned in this table
+
+| Out-of-band Tasks                | Priority Class      |  Reasons                                               |
+| ---------------------------------|---------------------|--------------------------------------------------------|
+| `recover-quorum-loss`            | Cluster-ops         | recovery from quorum loss involves etcd cluster, hence it's assign to `Cluster-ops` priority class. Any other operations of class: `Cluster-ops` can't succeed if there is no quorum present.   |
+| `on-demand snapshot`             | Cluster-ops         | taking a on-demand snapshot(full/delta) involves etcd cluster, hence it's assign to `Cluster-ops` priority class.                                       |
+| `on-demand snapshot-compaction`  | Backup-ops          | on-demand snapshot-compaction runs in a seprate pod, hence it don't depend on cluster.                                                                  |
+| `on-demand cluster maintenance`  | Cluster-ops         | on-demand etcd cluster maintenance depends on the health of etcd cluster, hence it's assign to `Cluster-ops` priority class.                            |
+| `copy-backup`                    | Backup-ops          | copy-backup tasks run seprately without affecting or required help from etcd cluster.                                                                   |
+
+1. Priority class `Cluster-ops` have 3 operator tasks(as of now)
+
+- recover-quorum-loss
+- on-demand cluster maintenance
+- on-demand snapshot
+
+2. Priority class `Backup-ops` have 2 operator tasks(as of now)
+
+- on-demand snapshot-compaction
+- copy-backup task
+
+Usage of priority class in deciding the precedence of simultaneous operator tasks:
+
+- All the operations in priorityClass: `Backup-ops` are independent from each other as well as independent from all operations under priorityClass:`Cluster-ops` as these `Backup-ops` can get execute seprately without depending on external factors like etcd cluster health, quorum etc.
```

Comment:
Somewhat (very) confusing to not find `OnDemandSnapshot` below the backup operations and instead under the less well defined (bucket) group of (generic) cluster operations. You probably did that so that you could make this statement here. This looks very contrived to me and I find it not a good decision (at all).